### PR TITLE
Feat: Add spring authentication

### PR DESCRIPTION
**Pull request by EduardoThums**

### CODE REVIEW

File: api/pom.xml [hunk 1]
```diff
@@ -46,6 +55,31 @@
...
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-oauth2-client</artifactId>
+            <version>${spring-security-oauth2.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.springframework.security</groupId>
+            <artifactId>spring-security-oauth2-jose</artifactId>
+            <version>${spring-security-oauth2.version}</version>
+        </dependency>
```
Comment (LeonardoBerlatto) [inline api/pom.xml]: Isn't this dependency for a different type of authentication?

File: api/src/test/java/help/desk/mobile/api/service/user/SaveUserServiceTest.java [hunk 2]
```diff
@@ -0,0 +1,151 @@
...
+		Assert.assertEquals(request.getName(), savedUserEntity.getName());
+		Assert.assertEquals(email, savedUserEntity.getUsername());
+		Assert.assertEquals(encodedPassword, savedUserEntity.getPassword());
+		Assert.assertEquals(email, savedUserEntity.getEmail());
+		Assert.assertEquals(cpf, savedUserEntity.getCpf());
+		Assert.assertEquals(phone, savedUserEntity.getPhone());
+		Assert.assertEquals(areaId, savedUserEntity.getAreaId());
+		Assert.assertEquals(ProfileType.DEFAULT, savedUserEntity.getProfile());
+
+		Assert.assertEquals(userId, response);
```
Comment (LeonardoBerlatto) [inline api/src/test/java/help/desk/mobile/api/service/user/SaveUserServiceTest.java:91]: I'd recommend using static import for assertEquals to improve clean code

File: api/pom.xml (hunk 1 above)
Comment (EduardoThums) [reply api/pom.xml]: Absolutely right, also I removed another unused auth dependency.

File: api/src/test/java/help/desk/mobile/api/service/user/SaveUserServiceTest.java (hunk 2 above)
Comment (EduardoThums) [reply api/src/test/java/help/desk/mobile/api/service/user/SaveUserServiceTest.java:91]: I don't think it's improve the code quality, for me it's good know who is the class of some static method.

**Pull request merged**